src/psyclone/psyir/transformations/nemo_sign_trans.py

```diff
@@ -0,0 +1,175 @@
+        self._operator_name = "SIGN"
+        self._classes = (BinaryOperation,)
+        self._operators = (BinaryOperation.Operator.SIGN,)
+
+    def apply(self, node, symbol_table, options=None):
+        '''Apply the SIGN intrinsic conversion transformation to the specified
+        node. This node must be an SIGN BinaryOperation. The SIGN
+        BinaryOperation is converted to the following equivalent
+        inline code:
+
+        R=SIGN(A,B) => R=ABS(B); if A<0.0 R=R*-1.0
+
+        This is implemented as a transform from:
+
+        R= ... SIGN(A,B) ...
+
+        to:
+
+        tmp_abs=B
+        IF (tmp_abs<0.0) res_abs=tmp_abs*-1.0 ELSE res_abs=tmp_abs
+        res_sign = res_abs
+        tmp_sign = A
+        if (tmp_sign<0.0) res_sign=res_sign*-1.0
+        R= ... res_x ...
+
+        where A and B could be an arbitrarily complex expressions and
+        where ABS is replaced with inline code by the NemoAbsTrans
+        transformation.
+
+        A symbol table is required as the NEMO api does not currently
+        contain a symbol table and one is required in order to create
+        temporary variables whose names do not clash with existing
+        code. This non-standard argument is also the reason why this
+        transformation is currently limited to the NEMO api.
+
+        This transformation requires the operation node to be a
+        descendent of an assignment and will raise an exception if
+        this is not the case.
+
+        :param node: a SIGN BinaryOperation node.
+        :type node: :py:class:`psyclone.psyGen.BinaryOperation`
+        :param symbol_table: the symbol table.
+        :type symbol_table: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :returns: 2-tuple of new schedule and memento of transform.
+        :rtype: (:py:class:`psyclone.nemo.NemoInvokeSchedule`, \
+                 :py:class:`psyclone.undoredo.Memento`)
+
+        '''
+        self.validate(node, symbol_table)
+
+        schedule = node.root
+        memento = Memento(schedule, self, [node, symbol_table])
+
+        oper_parent = node.parent
+        assignment = node.ancestor(Assignment)
+        # Create two temporary variables.
+        res_var = symbol_table.new_symbol_name("res_sign")
+        symbol_table.add(DataSymbol(res_var, DataType.REAL))
+        tmp_var = symbol_table.new_symbol_name("tmp_sign")
+        symbol_table.add(DataSymbol(tmp_var, DataType.REAL))
+
+        # Replace operator with a temporary (res_var).
+        oper_parent.children[node.position] = Reference(res_var,
+                                                        parent=oper_parent)
+
+        # res_var=ABS(B)
+        lhs = Reference(res_var)
+        rhs = UnaryOperation.create(UnaryOperation.Operator.ABS,
+                                    node.children[1])
```

Comment:
Should be `node.children[0]` I think.

Reply:
Agreed I've changed the code appropriately - essentially swapping A and B functionality.